src/wp-includes/class-wp-theme-json-resolver.php

```diff
@@ -543,8 +538,7 @@ public static function get_user_data() {
 
 		/** This filter is documented in wp-includes/class-wp-theme-json-resolver.php */
 		$theme_json   = apply_filters( 'wp_theme_json_data_user', new WP_Theme_JSON_Data( $config, 'custom' ) );
-		$config       = $theme_json->get_data();
-		static::$user = new WP_Theme_JSON( $config, 'custom' );
+		static::$user = $theme_json->get_theme_json();
```

Comment:
One thing I've noticed after reviewing the Gutenberg PR https://github.com/WordPress/gutenberg/pull/61262 is that this is different in Gutenberg: there, we do `$config['isGlobalStylesUserThemeJSON'] = true;` before the call to `WP_Theme_JSON`. We need to consolidate that logic.

Reply:
It looks like https://github.com/WordPress/wordpress-develop/pull/6616 will undo this change.

Reply:
@talldan w.r.t to keeping both the repos same, that change should be fine. But w.r.t performance we might need to find a better way to do that later on.